src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -57,6 +45,62 @@ public static StashCommentReport extractComments(String jsonBody) throws StashRe
     return result;
   }
 
+  public static StashComment extractComment(String jsonBody, String path, Long line) throws StashReportExtractionException {

```

Comment:
I would recomend to factorize the common part of the 2 methods in a 3rd, not public method. Something like:
  static StashComment extractComment(JSONObject jsonComment, String path, Long line) throws StashReportExtractionException {
    StashComment result = null;
    
      long id = (long) jsonComment.get("id");
      String message = (String) jsonComment.get("text");
      
      long version = (long) jsonComment.get("version");
      
      JSONObject jsonAuthor = (JSONObject) jsonComment.get("author");
      StashUser stashUser = extractUser(jsonAuthor.toJSONString());
      
      result = new StashComment(id, message, path, line, stashUser, version);

    return result;
  }
  
  

  public static StashComment extractComment(String jsonBody, String path, Long line) throws StashReportExtractionException {
    StashComment result = null;
    
    try {
      JSONObject jsonComment = (JSONObject) new JSONParser().parse(jsonBody);
      
      result = extractComment(jsonComment, path, line);

    } catch (ParseException e) {
      throw new StashReportExtractionException(e);
    }
    
    return result;
  }
  
  public static StashComment extractComment(String jsonBody) throws StashReportExtractionException {
    StashComment result = null;
    
    try {
      JSONObject jsonComment = (JSONObject) new JSONParser().parse(jsonBody);
      
      JSONObject jsonAnchor = (JSONObject) jsonComment.get("anchor");
      if (jsonAnchor == null) {
        throw new StashReportExtractionException("JSON Comment does not contain any \"anchor\" tag to describe comment \"line\" and \"path\"");
      }
      
      String path = (String) jsonAnchor.get("path");
        
      // can be null if comment is attached to the global file
      Long line = (Long) jsonAnchor.get("line");
      
      result = extractComment(jsonComment, path, line);

    } catch (ParseException e) {
      throw new StashReportExtractionException(e);
    }
    
    return result;
  }